### ISSUE TITLE
vbox: Cannot delete or start minikube if VM is deleted outside of minikube

### ISSUE THREAD

**Issue by woodcockjosh** [kubernetes/minikube#5509]
**The exact command to reproduce the issue**:
```bash
minikube start
😄  minikube v1.4.0 on Darwin 10.14.6
🔥  Creating virtualbox VM (CPUs=2, Memory=6500MB, Disk=20000MB) ...
^C
```
Cancelled the start with `Ctrl+c`

```bash
minikube start
```

Failed:
```bash
😄  minikube v1.4.0 on Darwin 10.14.6
💡  Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
🔄  Retriable failure: Error getting state for host: machine does not exist
```

Try to delete
```bash
minikube delete
```

Failed
```bash
💣  Unable to get the status of the cluster.
```

To fix the problem 
```bash
rm -r ~/.minikube/machines/minikube
```

You can reproduce the issue if you run `minikube start` and then delete the vm in the hypervisor GUI. 

**The output of the `minikube logs` command**:<details>

💣  command runner: getting ssh client for bootstrapper: Error dialing tcp via ssh client: dial tcp 127.0.0.1:22: connect: connection refused

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose

</details>

**The operating system version**:
Mac OS 10.14.6

**Comment by tstromberg** [comment]
Ugh. `minikube delete` is supposed to deal with missing VM's, but clearly it's not working as intended.

**Comment by nanikjava** [comment]
Stepping through the code when doing this noticed that the  executed command

> /usr/bin/VBoxManage showvminfo minikube --machinereadable

returned with the following error

> VBoxManage: error: Could not find a registered machine named 'minikube'
> VBoxManage: error: Details: code VBOX_E_OBJECT_NOT_FOUND (0x80bb0001), component VirtualBoxWrap, interface IVirtualBox, callee nsISupports
> VBoxManage: error: Context: "FindMachine(Bstr(VMNameOrUuid).raw(), machine.asOutParam())" at line 2621 of file VBoxManageInfo.cpp

and it is returned as part of the err object however the DeleteHost(..) is printing out as a generic error as 'Unable to get the status of the cluster.'

Is the requirement to report the error as what is returned to DeleteHost(..) or thinking of something else.

Happy to take this on if required.

**Comment by govargo** [comment]
Can I work on this?
I'll take.

**Comment by govargo** [comment]
I'm checking the code about this error.
I found following code section.

https://github.com/kubernetes/minikube/blob/master/pkg/minikube/cluster/cluster.go#L110
https://github.com/kubernetes/minikube/blob/master/pkg/minikube/cluster/cluster.go#L279

Now there is not "machine does not exist" error handling in `startHost` func and `DeleteHost` func.
So, I would add error handling when "machine does not exist" error occurs.

**Comment by govargo** [comment]
I'm now fixing this.

And I noticed the `minikube delete` failure is solved by #5654!
So, I'll fix error about `minikube start` failure.
